src/librustdoc/html/static/css/rustdoc.css

```diff
@@ -156,7 +156,9 @@ h1.fqn > .in-band > a:hover {
 	section hierarchies. */
 h2,
 .top-doc h3,
-.top-doc h4 {
+.top-doc h4,
+.sidebar .sidebar-elems h3:not(.sidebar-title),
```

Comment:
I find the `not` selectors are pretty hard to maintain and reason about. Can we do this with a positive selector instead? For instance, it looks like the headings we need underlines on here are nested under `.block`, so maybe we could do `.sidebar .block h3`?

Also it's not clear to me why both `.sidebar` and `.sidebar-elems` are needed in this selector.

Reply:
Except that the `.sidebar-title` are in `.block` as well. :3

It was more complicated than I expected to actually add this rule without breaking something.

Reply:
What about selecting the specific block types that get the underline treatment? Like:

```
.sidebar .block.struct h3, .sidebar .block.trait h3, .sidebar .block.type, ... {
```

Alternately, we could change the HTML structure slightly so everything after "Other items in ..." is nested under a specific class, like `.other-items`. Then we could make this selector `.sidebar .other-items h3`.

Reply:
I think I'd prefer to have a CSS class in the generated HTML, seems much simpler.